core/runtime/src/main/java/io/quarkus/runtime/logging/BannerFormatter.java

```diff
@@ -58,6 +59,13 @@ public String format(ExtLogRecord extLogRecord) {
 
     @Override
     public String getHead(Handler h) {
-        return bannerSupplier.get();
+        final String banner = bannerSupplier.get();
+        final Charset charset;
+        try {
+            charset = Charset.forName(h.getEncoding());
+            return charset.newEncoder().canEncode(banner) ? banner : "";
+        } catch (IllegalArgumentException ignored) {
+            return "";
```

Comment:
I was pulling my hair why this was resulting nothing printed...I added a `ignored.printStacktrace()` and getting this:

```
java.lang.IllegalArgumentException: Null charset name
	at java.base/java.nio.charset.Charset.lookup(Charset.java:455)
	at java.base/java.nio.charset.Charset.forName(Charset.java:526)
	at io.quarkus.runtime.logging.BannerFormatter.getHead(BannerFormatter.java:65)
	at org.jboss.logmanager.handlers.WriterHandler.writeHead(WriterHandler.java:114)
	at org.jboss.logmanager.handlers.WriterHandler.setWriter(WriterHandler.java:99)
	at org.jboss.logmanager.handlers.OutputStreamHandler.setWriter(OutputStreamHandler.java:99)
	at org.jboss.logmanager.handlers.ConsoleHandler.<init>(ConsoleHandler.java:123)
	at org.jboss.logmanager.handlers.ConsoleHandler.<init>(ConsoleHandler.java:100)
	at io.quarkus.runtime.logging.LoggingSetupRecorder.configureConsoleHandler(LoggingSetupRecorder.java:255)
	at io.quarkus.runtime.logging.LoggingSetupRecorder.initializeLogging(LoggingSetupRecorder.java:100)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit17.deploy_0(LoggingResourceProcessor$setupLoggingRuntimeInit17.zig:129)
	at io.quarkus.deployment.steps.LoggingResourceProcessor$setupLoggingRuntimeInit17.deploy(LoggingResourceProcessor$setupLoggingRuntimeInit17.zig:36)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:130)
	at io.quarkus.runtime.Application.start(Application.java:90)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:96)
	at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:60)
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:236)
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:39)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:131)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
	at io.quarkus.dev.DevModeMain.start(DevModeMain.java:104)
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:46)
```